### PR TITLE
feat: allow simulations to run in oracle dapp

### PR DESCRIPTION
**Pull request by gsteenkamp89**
## motivation

We want users to run simulations of transactions from oSnap proposals.
Allowing this in the Actions panel below the main call to action.

## screenshots
Screenshot 2024-04-01 at 21 40 01

Screenshot 2024-04-01 at 21 40 05
Screenshot 2024-04-01 at 21 40 09

### CODE REVIEW

File: src/components/Panel/OSnapTextData.tsx [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
-// Actual data could diverge from this, so we make everything partial.
-type SnapshotData = Partial<{
-  data: Partial<{
-    message: Partial<{
-      title: string;
-      space: string;
-    }>;
-  }>;
-  hash: string;
-}>;
```
Comment (gsteenkamp89) [inline src/components/Panel/OSnapTextData.tsx:21]: moved to a util file

File: src/constants/env.ts [hunk 2]
```diff
@@ -263,876 +267,80 @@ const Config = ss.object({
...
+    }
+    if (item === "PROVIDER") {
+      if (version === "SKINNY") {
+        const provider = {
+          source: "provider",
+          type: "Skinny Optimistic Oracle",
+          url: value,
+          chainId: parseInt(chainId),
+          address: getContractAddress({
+            chainId: parseInt(chainId),
```
Comment (gsteenkamp89) [inline src/constants/env.ts:317]: @daywiss I need your sign off on this. Adding config items for Sepolia proved extremely tedious and will be the case for any chain we add. So if we follow the structure outlined in the readme for these variables then this function will be able to parse the env file and find the correct config items.

Comment (gsteenkamp89) [reply src/constants/env.ts:317]: the structure must be `NEXT_PUBLIC_("SUBGRAPH" | "PROVIDER")_(VERSION | "SKINNY")_(CHAINID)`

File: src/types/snapshot.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  oSnap: {
+    safe: {
+      safeName: string;
+      safeAddress: string;
+      network: string;
+      moduleAddress: string;
+      transactions: Transaction[];
+    };
+  };
+};
```
Comment (gsteenkamp89) [inline src/types/snapshot.ts:62]: mostly imports from the snapshot app

**Review by daywiss — COMMENTED** [review]
im not seeing the simulation button on mainnet in a current osnap request
https://optimistic-oracle-dapp-v2-ik3vnpc0x-uma.vercel.app/?transactionHash=0xec29cff210f4a3e7c4ac614f6fe71f2e599058e419f395a86a8436615ce5ea92&eventIndex=168

File: src/constants/tenderly.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const SIMULATION_ENDPOINT =
+  "https://ethereum-api-read-prod-77jg7zf4ea-ue.a.run.app/osnap/simulate";
```
[the remark this answers is not in the document]
Comment (gsteenkamp89) [reply src/constants/tenderly.ts:2]: done

**Review by daywiss — APPROVED** [review]

**Pull request merged**